co_sim_io/includes/model_part.hpp

```diff
@@ -266,10 +287,19 @@ class CO_SIM_IO_API ModelPart
 
     void Clear();
 
+protected:
+    friend class std::unique_ptr<ModelPart>;
+    ModelPart(const std::string& I_Name, const bool InitInternalModelParts);
```

Comment:
This is a bit of a hack as I don't want to expose this CTor

Basically the idea is that I want to initialize the internal ModelParts in the CTor, but this would results in an infinite recursion
Hence I pass a flag to prevent this